### ISSUE TITLE
Change exception to warning in AUC metric

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#3672]
**Is your feature request related to a problem? Please describe.**
Currently, when the input data is all 0 or all 1, AUC will raise an error:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/rocauc.py#L69

@SachidanandAlle , @madil90 , and our internal QAs all encountered this error in classification tasks before, especially when **verifying the pipeline with very few samples**. For example, it's easy to trigger the exception in chest X-ray classification task as the task has 15 classes, easy to get all 0 or all 1 samples for a class.

I would like to change the exception to warning and skip the AUC computation for this case, or add a flag to control whether a warning or exception?
@ericspod @wyli @rijobro Do you guys have any other concerns?

Thanks in advance.

**Comment by wyli** [comment]
yes, I think we can follow the scikit-learn https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/metrics/_ranking.py#L991

**Comment by ericspod** [comment]
The warning makes sense but we should have a more informative error message, that is if the input is all one value say explicitly that's what the issue is and will produce bad results.

**Comment by Nic-Ma** [comment]
> The warning makes sense but we should have a more informative error message, that is if the input is all one value say explicitly that's what the issue is and will produce bad results.

Hi @ericspod ,

As the previous PR had been merged, I submitted a follow up to address your comments:
https://github.com/Project-MONAI/MONAI/pull/3676
And added more unit tests to cover all the cases.
Could you please help review it?

Thanks in advance.

**Comment by ericspod** [comment]
Done, thanks!
